clients/pd-send/main.go

```diff
@@ -0,0 +1,109 @@
+/*
+ *  This is a basic client that can plug into Nagios.
+ */
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"hash/crc32"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/streadway/amqp"
+)
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Fatalf("%s: %s", msg, err)
+	}
+}
+
+func getEnvWithDefault(env string, default_value string) string {
+	var v = os.Getenv(env)
+	if len(v) == 0 {
+		v = default_value
+	}
+	return v
+}
+
+func makeBody() (string, []byte) {
+	m := make(map[string]interface{})
```

Comment:
This works, normally expect to see structs leveraged for JSON marshalling and clarity, e.g.

```go
type body struct {
  routing_key string // Could use struct tags but not necessary as naming is consistent.
  event_action string
  payload payload
}

...

m = body{
  routing_key: routing_key
}
```

And similar for `payload`.

Reply:
I considered that but the V2 map is quite extensive, mostly unused, and maybe apart from some conciseness there's no advantage in keeping the stuff around as a struct. A simple map does the job well enough (and probably mildly simpler to understand than having all sorts of pointers into a preallocated struct with `flag.StringVar` etcetera). I guess it's a cointoss.

Reply:
The potential complexity of the map is good justification, thanks for clarifying!